File: src/main/org/nlogo/prim/threed/_worlddepth.java
package org.nlogo.prim.threed;

import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _worlddepth extends Reporter
{
	@Override public Syntax syntax()
	{
		return Syntax.reporterSyntax( Syntax.TYPE_NUMBER ) ;
	}
	@Override public Object report( Context context ) 
	{
		return report_1( context ) ;
	}
	public double report_1( Context context ) 
	{
		return ( (org.nlogo.agent.World3D) world ).worldDepth() ;
	}
}

File: src/main/org/nlogo/prim/threed/_worlddepth.scala
package org.nlogo.prim.threed

import org.nlogo.agent.World3D
import org.nlogo.nvm.{ Context, Reporter, Syntax }

class _worlddepth extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.TYPE_NUMBER)
  override def report(context: Context) =
    java.lang.Double.valueOf(world.asInstanceOf[World3D].worldDepth)
}
